### ISSUE TITLE
error when running macro on keyence tiff files

### ISSUE THREAD

**Issue by moj852** [rentazillla/KeyenceImageMergerMacro#1]
I have a folder of ~100 tiff files exported from a zstack. 4 channels. Each zstack image has _CH1 _CH2 at the end. When I run this macro it gives me the following error:
![image](https://github.com/rentazillla/KeyenceImageMergerMacro/assets/134652679/19fa9641-9ff7-4f27-8783-55bd5350cd15)

**Comment by rentazillla** [comment]
Do you have the images saved with all four channels in the same image? This macro is designed to open the four channels as separate images

**Comment by moj852** [comment]
Yep all the images are separate. See below for a screenshot of the image names

![image](https://github.com/rentazillla/KeyenceImageMergerMacro/assets/134652679/0bd1c6f4-f6d1-4d70-b887-9ccdbeaa254c)

**Comment by rentazillla** [comment]
It looks like the Z stack is separated in each channel. You should first create stacks in ImageJ such that each _CHX file contains all the Z positions for that channel. You can use a macro like this to automate that if the input folder contains all the channels for a single sample (perhaps 01L-M-20x if I am interpreting your file naming correctly): 

```
#@ File (label = "Input directory", style = "directory") input
#@ File (label = "Output directory", style = "directory") output

prefix = "01L-M-20x";
suffix = newArray("_A", "_B", "_C", "_D");
channels = newArray("_CH1", "_CH2", "_CH3", "_CH4");
for (j = 0; j < channels.length; j++) {
	setOption("WaitForCompletion", true);
	// Get all the images for all channels
	args = "open=" + input;
	args += " file=" + channels[j];
	args += " sort";
	run("Image Sequence...", args);
	run("8-bit");
	number = getList("image.titles");
	id = getTitle();
	selectImage(id);
	rename(prefix + suffix[j] + channels[j]);
	save(output + File.separator + prefix + suffix[j] + channels[j] + ".tif");
}
```
